source/h2_decoder.c

```diff
@@ -752,6 +770,15 @@ static int s_state_fn_frame_settings_i(struct aws_h2_decoder *decoder, struct aw
     /* An endpoint that receives a SETTINGS frame with any unknown or unsupported identifier MUST ignore that setting.
      * RFC-7540 6.5.2 */
     if (id >= AWS_H2_SETTINGS_BEGIN_RANGE && id < AWS_H2_SETTINGS_END_RANGE) {
+        /* check the value meets the settings bounds */
+        if (value < aws_h2_settings_bounds[id][0] || value > aws_h2_settings_bounds[id][1]) {
+            DECODER_LOGF(ERROR, decoder, "The %dth value of SETTING frame is invalid", id);
```

Comment:
log the id and value of the invalid setting. Thats much more informative than knowing it was the Nth entry in the frame